### DISCUSSION TITLE
vuepress-plugin-components installation breaks my docs

### DISCUSSION THREAD

**Discussion by ghost** [vuepress-theme-hope/vuepress-theme-hope#2538] [Q&A] [answered]
When I installed the plugin, my documents became unavailable - the webpage will be a blank white. I have created a document outlining the steps I have taken. What further action should I take in order to get the plugin to work with my documents?

<details>
<summary>Running the current master this is what you see when you start the development server:
</summary>

```
npm start

> wasabidoc@0.1.0 prestart
> rm -rf $npm_package_config_dist


> wasabidoc@0.1.0 start
> vuepress dev docs

warning an anonymous plugin or theme was detected and ignored
warning an anonymous plugin or theme was detected and ignored
warning an anonymous plugin or theme was detected and ignored
✔ Initializing and preparing data - done in 986ms

  vite v3.1.8 dev server running at:

  ➜  Local:   http://localhost:8080/
```
</details>

<details>
<summary>Now install the plugin:</summary>

```
$ npm i -D vuepress-plugin-components@next

added 326 packages, removed 2 packages, changed 24 packages, and audited 795 packages in 21s

78 packages are looking for funding
  run `npm fund` for details

6 vulnerabilities (2 moderate, 4 high)

To address issues that do not require attention, run:
  npm audit fix

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
```
</details>

<details>
<summary>Running the current master AFTER INSTALLING PLUGIN this is what you see when you start the development server:
</summary>

```
$ npm start

> wasabidoc@0.1.0 prestart
> rm -rf $npm_package_config_dist


> wasabidoc@0.1.0 start
> vuepress dev docs

warning an anonymous plugin or theme was detected and ignored
warning an anonymous plugin or theme was detected and ignored
warning an anonymous plugin or theme was detected and ignored
✔ Initializing and preparing data - done in 1.19s

  vite v4.0.0 dev server running at:

  ➜  Local:   http://localhost:8080/
```
</details>

In short, the difference I can see before and after installing the plugin is:
*  vite v3.1.8 dev server running at
*  vite v4.0.0 dev server running at

Could this be causing the issue? What do you recommend?

**Comment by Mister-Hope** [comment, marked as answer]
See release notes. Each version has a compatable version

**Comment by ghost** [comment]
Thank you for your quick response. I have another question regarding updating the config.ts file. I changed the version to `2.0.0-beta.112` and it will render after installing. But when I edit the config.ts file and try to render my docs website, I get the following error:
`TypeError: Cannot read properties of undefined (reading 'forEach')`

I did some research and found out that this error is likely due to an incorrect configuration of the plugin. The source said to make sure you are correctly setting the options in the config.ts file and that all of the required parameters are provided. **So can you look at how I import the plugin below to see if this may be the issue**?

At the top of the config.ts file I import the plugin:
`import { componentsPlugin } from "vuepress-plugin-components"`

Then in the plugins section of the file I add the componentsPlugin below an existing searchPlugin (here is that snippet):

```
...
...
searchPlugin({
            // getExtraFields: (page) => page.frontmatter.tags,
            maxSuggestions: 15,
            hotKeys: ['s', '/'],
            locales: {
                '/': {
                    placeholder: 'Search...',
                }
            }
        }),
componentsPlugin({
        options: {
            components: {
                youtube: true
            }
       }
})
...
...
```

Do you see what could be the issue? Note that this is regarding the discussion at: https://github.com/vuepress/vuepress-next/discussions/1182

**Reply by Mister-Hope** [reply]
The plugin docs is stick to the latest version. If you use older version, you may check them at repo via https://github.com/vuepress-theme-hope/vuepress-theme-hope/tree/v2.0.0-beta.112/docs/components/src

**Reply by Mister-Hope** [reply]
We encourage you to use latest version of vuepress@next as older version may contain issues.

**Reply by ghost** [reply]
> We encourage you to use latest version of vuepress@next as older version may contain issues.

I don't quite understand. When I download the latest version with `npm i -D vuepress-plugin-components@next' there seems to be a conflict and I am unable to see any of my website - it's completely blank white. But when I download an older version, it seems fine.

And is the following the correct syntax in the config.ts file for adding the YouTube component? I'm really new to Type Script and vuePress for that matter.
```
componentsPlugin({
              components: [
                "YouTube",
              ],
        })
```

Thanks for your help!

**Reply by Mister-Hope** [reply]
If your other plugins support vuepress2 beta.58, you can upgrade to `"2.0.0-beta.58"`.

I hate talking without actual codes. You should provide your repo link or some reproduction to let others tell you what's wrong.

VuePressThemeHope is using this plugin in all versions, and only <10 versions of all the 200+ since v2 is broken, and version like that is warned in release page.

**Reply by Mister-Hope** [reply]
```
componentsPlugin({
              components: [
                "YouTube",
              ],
        })
```

This should be correct if your editor support ts and did not warn you anything, that's why we use ts and add a lot of js docs.

**Reply by Mister-Hope** [reply]
TBH, I can remember history changes, but I am not able to remember which version are those changes applied. Old version config should be that way as you are asking, but I am not sure if we support all the props docs mentions.
